micromagneticdata/data.py

```diff
@@ -169,7 +169,11 @@ def __getitem__(self, item):
         Drive(...)
 
         """
-        return md.Drive(name=self.name, number=item % self.n, dirname=self.dirname)
+        return md.Drive(
+            name=self.name,
+            number=[i for i in range(self.n)][item],
```

Comment:
`number=range(self.n)[item]`

Reply:
Good point! I was under the impression that `range()` is genexp.

Reply:
I see.